### ISSUE TITLE
Negative Mean Rewards [help wanted]

### ISSUE THREAD

**Issue by Serafiniert** [Unity-Technologies/ml-agents#182]
I've successfully trained the 3D Ball example. The results with unaltered hyperparameters were a mean reward of 50-80 initially, but decreased to 5-20. The exported .bytes would result was decent when used as internal brain in Unity. (A few balls dropped, but quickly all platforms could balance them just fine).

Now I've tried the Tennis example. But training it had some strange mean rewards:

> Step: 10000. Mean Reward: -0.019164564192888086. Std of Reward: 0.0786144832296523.
> Step: 20000. Mean Reward: -0.02087563569116967. Std of Reward: 0.08042892856530069.
> Step: 30000. Mean Reward: -0.020126928243850463. Std of Reward: 0.08035407475490902.
> Step: 40000. Mean Reward: -0.019409300775105907. Std of Reward: 0.07859521812418054.
> Step: 50000. Mean Reward: -0.019506338196603683. Std of Reward: 0.07840685426664472.

I tried to change some parameters (num_layers, epsilon, buffer_size, learning_rate, hidden_units, batch_size) to see if changing any of them will impact the results of the training. But every mean reward is negative. Why is the output so different from 3D Ball? How can I manipulate them?

**Comment by awjuliani** [comment]
Hi @Serafiniert,

In order to successfully train Tennis, you will need to run the training for 3-5 million steps, rather than just 50,000. The reward should eventually become positive.
